tower-limit/src/rate/service.rs

```diff
@@ -65,7 +65,9 @@ where
         match self.state {
             State::Ready { .. } => return self.inner.poll_ready().map_err(Into::into),
             State::Limited(ref mut sleep) => {
-                try_ready!(sleep.poll());
+                if let Async::NotReady = sleep.poll()? {
+                    tracing::trace!("rate limited exceeded; sleeping.");
```

Comment:
@lukesteensen I like this better, what do you think?

Reply:
A little late on this review, but I think this actually broke the whole thing :smile: 

Before, `try_ready!` would return `NotReady` if we still needed to sleep. Now we're just checking if we should still be sleeping, logging that we should, and continuing on and marking ourselves `Ready`.